plugins/modules/archive.py

```diff
@@ -542,6 +550,33 @@ def _get_checksums(self, path):
         return checksums
 
 
+class ReproducibleTGZFile(tarfile.TarFile):
+    def __init__(
+        self, name=None, mode=None, compresslevel=-1, fileobj=None, mtime=None, **kwargs
+    ):
+        if fileobj is None:
+            fileobj = open(name, mode + "b")
+
+        try:
+            # output filename intentionally empty exclude it from gzip header
+            gzipfileobj = gzip.GzipFile("", mode, compresslevel, fileobj, mtime)
+        except Exception:
+            fileobj.close()
+            raise
+
+        # Allow GzipFile to close fileobj as needed
+        gzipfileobj.myfileobj = fileobj
+
+        try:
+            super(ReproducibleTGZFile, self).__init__(mode=mode, fileobj=gzipfileobj, **kwargs)
+        except Exception:
+            gzipfileobj.close()
+            raise
+
+        # Allow TarFile to close GzipFile as needed
+        self._extfileobj = False
```

Comment:
Same here.

How about overwriting `close()` as well and closing everything in there? I think that would be cleaner.